### PR TITLE
Do not store the value, only store the labels and global reference id…

### PR DESCRIPTION
**Pull request by mattdurham**
#### PR Description

Instead of storing the whole flowmetric, which includes the value. We only cache the global cache id and the labels. Storing the global ref id saves a lookup and in the simple benchmark saves about 15% . This resolves the issue where the first value is locked in and always reused.

### CODE REVIEW

File: component/prometheus/relabel/relabel.go [hunk 1]
```diff
@@ -154,12 +159,26 @@ func (c *Component) clearCache() {
...
 }
 
 func (c *Component) addToCache(originalID uint64, fm *prometheus.FlowMetric) {
 	c.cacheMut.Lock()
 	defer c.cacheMut.Unlock()
 
-	c.cache[originalID] = fm
+	if fm == nil {
+		c.cache[originalID] = nil
+		return
```
Comment (rfratto) [inline component/prometheus/relabel/relabel.go:171]: Should this delete the entry instead of setting it to nil?

**Review by rfratto — COMMENTED** [review]
Can we add a test for the behavior we experienced? That's the type of thing I'd like to catch in a unit test to avoid regressions.

**Comment by mattdurham** [comment]
Will add some more tests.

File: component/prometheus/relabel/relabel.go (hunk 1 above)
Comment (mattdurham) [reply component/prometheus/relabel/relabel.go:171]: This handles the case of dropping a metric entirely.

Comment (rfratto) [reply component/prometheus/relabel/relabel.go:171]: Ah, got it 👍

**Review by rfratto — APPROVED** [review]

**Pull request merged**